Scram only compile

swayand opened the pull request:
Changes are made, which allows us to compile everything with scram. 

This includes also the update of 
KappaTools
Artus
HiggsAnalysis-KITHiggsToTauTau
Excalibur 

I also made some basic cross checks, since most of travis test will fail. For the Test i succefully run HiggsToTauTauAnalysis.py and higgsplot.py. with some test files. Also the skimming worked for me (Additional i removed some double declaration of basics root classes)

If you want to run a test for HiggsTauTau you can run following:
wget https://raw.githubusercontent.com/cms-analysis/HiggsAnalysis-KITHiggsToTauTau/scramonly/scripts/checkout_packages_scramonly.sh
source checkout_packages_scramonly.sh

i hope i covered most of the programs which needed modifications in the linking step. If you found something which is not working anymore, i can help you to fix it. 

Cheers
Stefan

Commit:
First test commit, with scram only working

Commit:
Merge branch 'master' of github.com:swayand/Kappa

Commit:
next try

Commit:
get rid of double classes count

Commit:
moving ini script kappa.xml needed by HiggsTauTau from test folder to script folder (test is not recomended for needed files)

Commit:
fixing variables

thomas-mueller commented:
Hallo Stefan,

This idea is very welcome! Thanks for your work here.

For me it is not so clear, why many binaries have been removed from being compiled. Another question is related with the removal of loading the Kappa library in the python scripts: Is this now automatically available via CMSSW?

I think, we should wait for a comment from Fred before merging this, because so much code has been deleted, where I fear nobody else has a complete overview. And about the failing tests others have to judge.

Cheers,
Thomas

swayand commented:
Hi Thomas,

thanks for the feedback. Previous we compiled Kappa or better the Kappa Data formats twice, once with scram and per hand with make. But if the cmsenv is set then it knows about the compiled libs, so you only need to tell where they are compiled (meaning replace in all Buildfiles.xml   <Flags LDFLAGS="-Lsrc/Kappa/lib -lKappa" />  =>  <use name="Kappa/DataFormats" />) 

Concerning the removed code, this all belongs to the Kappa/DataFormats/test folder, which was deleted by me to get sure that this is  not used by any program anymore. Which triggers a more academic question: For me it is not good if software packages relies on code which is stored in a folder named test. 
In this folder also some Debug functions was declared, so why not put them in the regular place? 

The other think i have deleted was the classes.UP script, which automatically generates the files:
- DataFormats/interface/KDebug.h 
- DataFormats/interface/Kappa.h 
- classes.h
- classes_def.xml 
- LinkDef.h 
  But depends on the input DataFormats/test/KDebug.cpp , so if people use this script one can recover it, by adding again DataFormats/test/KDebug.cpp. This was also the major concern Fred had about this commit. 

For the test Georg offers me help me to recover them. By the way, does anybody use CMSSW_7_4_X or below? 

Cheers
Stefan

claria commented:
Hi Stefan,

if you push your fork of Kappa as separate branch in this repository i can fix the tests. 

Cheers,
Georg

fcolombo commented:
Hi,
removing completely the "test" folder also removed the Kappa test script that Joram created some time ago and which produces summary outputs like:
http://www-ekp.physik.uni-karlsruhe.de/~fcolombo/kappa/test/current/result.html

Is this intended and/or is the group not interested anymore in this tool?
This is a genuine question. At the moment I'm still "administering" the tool, but since almost nothing has been changed compared to what Joram created, this task simply consists in:
-) checking that the script runs in a cronjob and that the output are correctly copied to the webpage
-) occasionally ping people responsible for commits which cause the test to fail (usually this is easily traced back to release-dependent commits)

I consider the tool to be nice and a good example of the power and flexibility of Kappa in working under different CMSSW releases (the summary page is also a good propaganda plot) but since I'm basically not anymore in the group I don't want to weight in with my opinion, so the decision is up to you guys.
If you want to keep this, it is obvious that, on a timescale of ~O(month), someone else needs to keep and eye on this and subscribe for the easy two tasks that I mentioned above.

FredStober commented:
> For me it is not good if software packages relies on code which is stored in a folder named test. 

The reason back then was that if you used **any** other directory name, scram automatically linked the source code into the CMSSW module library.
Previously, this lead to multiple root dictionaries (LCG vs. rootcint) being linked into the same file, which confused root during writing the tuples. `So simply put: The directory is called test because it's the only one that scram doesn't include in the default target.`

I guess the real problem is (I think i stated it a long time ago already): Why are you interested in the rootcint dictionary that is intended for standalone, non-scram Kappa applications, when you are in the SCRAM environment that provides the LCG dictionary that is used by CMSSW?
In an ideal world you shouldn't be interested in any compiled file in there - (well at least after moving KDebug.cpp from there, but that file is just for debugging ...) - since it provides just all the things that are needed to use Kappa from a basic root environment.

Anyway - you have some setup now that works for you so thats fine.

swayand commented:
Thank you for all your replies, i think there were very helpful. 

@FredStober 
I can rememberer that in the past SCRAM tries to compile every think, but now (i guess its about 3 or more years) it only goes true the directories src and interface. If you want that a folder (or file) gets compiled, you have to put an Buildfile.xml in it. So there is no need to put all into the test folder. The aim for this commits are indeed to disentangled the SCRAM and MAKE world. And maintain only SCRAM functionality. 

@fcolombo 
The point fabio mentioned is a good one, but only partly related to this topic. My suggestion would be to move such stuff into a monitoring (or something similar) folder, and give it a better name (it was my mistake but if something is called test/test.py i usually don't expect important code). But lets discuss this in an separate issue. 

To conclude on this, i will recover the test folder in the corresponding branch (and so also the classes.UP script). Ask our main analyzer (Raphael, Rene, Alexander) in our tomorrows meeting if they have any concerns, and if not merge everything into the master. 

Cheers

thomas-mueller commented:
Hi Stefan,

you said, that you only removed Kappa testing binaries from beeing compiled, but this is not true: https://github.com/KappaAnalysis/KappaTools/pull/10/files, https://github.com/artus-analysis/Artus/pull/50/files. All the BuildFile.xml changes in KappaTools and Artus are related with executables, where I do not see any reason why we should not continue in compiling them.

Then another point: I am not sure, whether a merge can be so fast, mainly because Raphael, Rene and Alexander are not the only main users. Do you have feedback from the Z+jet group? If I remember correctly, they compile theie analysis independently from CMSSW. This would mean that the make-features must be kept working. For this I would expect changes in Makefiles.

Regards,
Thomas

swayand commented:
Hi Thomas,

so I have added the compile (and/or linking) of all binaries again. If something is missing, its only one line of code which one has to add. The problem was that some of the binaries needed the KDebug.cpp. Therefore i have moved them from the DataFormats/test to DataFormats/src so that it gets compiled automatically. 

Concerning the Z+jet group. I started the pull request for this group, since we are planning to move to scram compile, with the benefit that one only needs a running CMSSW environment and all developments from the Higgs group also works out of the box for the z+jet group. So the OK from the Higgs group is the only show stopper for this pull request. 

Best, 
Stefan

thomas-mueller commented:
Hi Stefan, all,

Compiling Kappa with scram needs a bit more testing. I made two commits in Kappa and KappaTools. In order to not rely on the make outputs I propose to delete the Kappa libraries manually:

```
rm Kappa/DataFormats/test/libKappa.*
rm -r Kappa/lib
```

If you stil have the Makefile, `make clean` can also be a good thing to do.

After these changes, which only remove stuff related to make, it does not work for me. Feedback and help is very welcome. (I am actually thinking about reverting the merges and my commits, until the testing is finished.)

Regards,
Thomas

swayand commented:
Sorry, but i don't get your point.

You don't need to run any make command. Just check out a new version, and run scram only from scratch. 

Cheers
Stefan

thomas-mueller commented:
I discussed already some details in the KIT skype chat with Raphal and Alex. It seems, that batch submission only works for people who did not clean up the make compilation. So they compiled with scram, but the libraries created previously by make were still there and taken for the batch submission. (At least this is my current understanding.) It might make sense to test it (running locally and on a batch system) in a CMSSW environment set up from scratch.

thomas-mueller commented:
Data source: default
input_line_219:65:10: fatal error: 'Kappa/DataFormats/interface/Hash.h' file not found
#include "Kappa/DataFormats/interface/Hash.h"
         ^
Error in TInterpreter::AutoParse: Error parsing payload code for class KElectron with content:

#ifndef G__VECTOR_HAS_CLASS_ITERATOR
  #define G__VECTOR_HAS_CLASS_ITERATOR 1
#endif
#ifndef CMS_DICT_IMPL
  #define CMS_DICT_IMPL 1
#endif
#ifndef _REENTRANT
  #define _REENTRANT 1
#endif
#ifndef GNUSOURCE
  #define GNUSOURCE 1
#endif
#ifndef __STRICT_ANSI__
  #define **STRICT_ANSI** 1
#endif
#ifndef GNU_GCC
  #define GNU_GCC 1
#endif
#ifndef _GNU_SOURCE
  #define _GNU_SOURCE 1
#endif
#ifndef CMSSW_GIT_HASH
  #define CMSSW_GIT_HASH "CMSSW_7_4_7"
#endif
#ifndef PROJECT_NAME
  #define PROJECT_NAME "CMSSW"
#endif
#ifndef PROJECT_VERSION
  #define PROJECT_VERSION "CMSSW_7_4_7"
#endif
#ifndef BOOST_SPIRIT_THREADSAFE
  #define BOOST_SPIRIT_THREADSAFE 1
#endif
#ifndef PHOENIX_THREADSAFE
  #define PHOENIX_THREADSAFE 1
#endif
#ifndef CMSSW_REFLEX_DICT
  #define CMSSW_REFLEX_DICT 1
#endif

#define _BACKWARD_BACKWARD_WARNING_H
/\* This file is autogenerated by classes.UP. Do not edit!

```
Copyright (c) 2010 - 2015, All Rights Reserved
The authors are listed in the AUTHORS file.

ROOT dictionary that Kappa uses for reading the Kappa skim

    http://root.cern.ch/drupal/content/selecting-dictionary-entries-linkdefh
    http://root.cern.ch/drupal/content/interacting-shared-libraries-rootcint
```

*/

#ifndef KDATAFORMAT_CLASSES_H
#define KDATAFORMAT_CLASSES_H

#define G__DICTIONARY

#include "Kappa/DataFormats/interface/Hash.h"
#include "Kappa/DataFormats/interface/KBasic.h"
#include "Kappa/DataFormats/interface/KElectron.h"
#include "Kappa/DataFormats/interface/KInfo.h"
#include "Kappa/DataFormats/interface/KJetMET.h"
#include "Kappa/DataFormats/interface/KLepton.h"
#include "Kappa/DataFormats/interface/KMuon.h"
#include "Kappa/DataFormats/interface/KParticle.h"
#include "Kappa/DataFormats/interface/KPhoton.h"
#include "Kappa/DataFormats/interface/KRoot.h"
#include "Kappa/DataFormats/interface/KTau.h"
#include "Kappa/DataFormats/interface/KTrack.h"

namespace
{
        struct dictionary
        {
                RMFLV dict_RMFLV;
                RMFLV_Store dict_RMFLV_Store;
                RMDLV dict_RMDLV;
                RMDLV_Store dict_RMDLV_Store;
                KEventInfo dict_KEventInfo;
                KGenEventInfo dict_KGenEventInfo;
                KBasicJet dict_KBasicJet;
                KBasicJets dict_KBasicJets;
                KBasicTau dict_KBasicTau;
                KBasicTaus dict_KBasicTaus;
                KBeamSpot dict_KBeamSpot;
                KBeamSpots dict_KBeamSpots;
                KCaloJet dict_KCaloJet;
                KCaloJets dict_KCaloJets;
                KElectron dict_KElectron;
                KElectrons dict_KElectrons;
                KLV dict_KLV;
                KLVs dict_KLVs;
                KGenJet dict_KGenJet;
                KGenJets dict_KGenJets;
                KGenParticle dict_KGenParticle;
                KGenParticles dict_KGenParticles;
                KGenPhoton dict_KGenPhoton;
                KGenPhotons dict_KGenPhotons;
                KGenTau dict_KGenTau;
                KGenTaus dict_KGenTaus;
                KHit dict_KHit;
                KHits dict_KHits;
                KJet dict_KJet;
                KJets dict_KJets;
                KL1Muon dict_KL1Muon;
                KL1Muons dict_KL1Muons;
                KLepton dict_KLepton;
                KLeptons dict_KLeptons;
                KLeptonPair dict_KLeptonPair;
                KLeptonPairs dict_KLeptonPairs;
                KMET dict_KMET;
                KMETs dict_KMETs;
                KMuon dict_KMuon;
                KMuons dict_KMuons;
                KMuonTriggerCandidate dict_KMuonTriggerCandidate;
                KMuonTriggerCandidates dict_KMuonTriggerCandidates;
                KParticle dict_KParticle;
                KParticles dict_KParticles;
                KPFCandidate dict_KPFCandidate;
                KPFCandidates dict_KPFCandidates;
                KPhoton dict_KPhoton;
                KPhotons dict_KPhotons;
                KPileupDensity dict_KPileupDensity;
                KTau dict_KTau;
                KTaus dict_KTaus;
                KExtendedTau dict_KExtendedTau;
                KExtendedTaus dict_KExtendedTaus;
                KTaupairVerticesMap dict_KTaupairVerticesMap;
                KTaupairVerticesMaps dict_KTaupairVerticesMaps;
                KTrack dict_KTrack;
                KTracks dict_KTracks;
                KTriggerObjects dict_KTriggerObjects;
                KVertex dict_KVertex;
                KVertices dict_KVertices;
                KFilterSummary dict_KFilterSummary;
                KHCALNoiseSummary dict_KHCALNoiseSummary;
                KTrackSummary dict_KTrackSummary;
                KVertexSummary dict_KVertexSummary;
                KLumiInfo dict_KLumiInfo;
                KGenLumiInfo dict_KGenLumiInfo;
                KDataLumiInfo dict_KDataLumiInfo;
                KElectronMetadata dict_KElectronMetadata;
                KFilterMetadata dict_KFilterMetadata;
                KMuonMetadata dict_KMuonMetadata;
                KGenEventInfoMetadata dict_KGenEventInfoMetadata;
                KJetMetadata dict_KJetMetadata;
                KTauMetadata dict_KTauMetadata;
                KTriggerObjectMetadata dict_KTriggerObjectMetadata;
                KProvenance dict_KProvenance;
        };
}

#endif

#undef  _BACKWARD_BACKWARD_WARNING_H

input_line_220:65:10: fatal error: 'Kappa/DataFormats/interface/Hash.h' file not found
#include "Kappa/DataFormats/interface/Hash.h"
         ^
Error in TInterpreter::AutoParse: Error parsing payload code for class KMuon with content:

#ifndef G__VECTOR_HAS_CLASS_ITERATOR
  #define G__VECTOR_HAS_CLASS_ITERATOR 1
#endif
#ifndef CMS_DICT_IMPL
  #define CMS_DICT_IMPL 1
#endif
#ifndef _REENTRANT
  #define _REENTRANT 1
#endif
#ifndef GNUSOURCE
  #define GNUSOURCE 1
#endif
#ifndef __STRICT_ANSI__
  #define **STRICT_ANSI** 1
#endif
#ifndef GNU_GCC
  #define GNU_GCC 1
#endif
#ifndef _GNU_SOURCE
  #define _GNU_SOURCE 1
#endif
#ifndef CMSSW_GIT_HASH
  #define CMSSW_GIT_HASH "CMSSW_7_4_7"
#endif
#ifndef PROJECT_NAME
  #define PROJECT_NAME "CMSSW"
#endif
#ifndef PROJECT_VERSION
  #define PROJECT_VERSION "CMSSW_7_4_7"
#endif
#ifndef BOOST_SPIRIT_THREADSAFE
  #define BOOST_SPIRIT_THREADSAFE 1
#endif
#ifndef PHOENIX_THREADSAFE
  #define PHOENIX_THREADSAFE 1
#endif
#ifndef CMSSW_REFLEX_DICT
  #define CMSSW_REFLEX_DICT 1
#endif

#define _BACKWARD_BACKWARD_WARNING_H
/\* This file is autogenerated by classes.UP. Do not edit!

```
Copyright (c) 2010 - 2015, All Rights Reserved
The authors are listed in the AUTHORS file.

ROOT dictionary that Kappa uses for reading the Kappa skim

    http://root.cern.ch/drupal/content/selecting-dictionary-entries-linkdefh
    http://root.cern.ch/drupal/content/interacting-shared-libraries-rootcint
```

*/

#ifndef KDATAFORMAT_CLASSES_H
#define KDATAFORMAT_CLASSES_H

#define G__DICTIONARY

#include "Kappa/DataFormats/interface/Hash.h"
#include "Kappa/DataFormats/interface/KBasic.h"
#include "Kappa/DataFormats/interface/KElectron.h"
#include "Kappa/DataFormats/interface/KInfo.h"
#include "Kappa/DataFormats/interface/KJetMET.h"
#include "Kappa/DataFormats/interface/KLepton.h"
#include "Kappa/DataFormats/interface/KMuon.h"
#include "Kappa/DataFormats/interface/KParticle.h"
#include "Kappa/DataFormats/interface/KPhoton.h"
#include "Kappa/DataFormats/interface/KRoot.h"
#include "Kappa/DataFormats/interface/KTau.h"
#include "Kappa/DataFormats/interface/KTrack.h"

namespace
{
        struct dictionary
        {
                RMFLV dict_RMFLV;
                RMFLV_Store dict_RMFLV_Store;
                RMDLV dict_RMDLV;
                RMDLV_Store dict_RMDLV_Store;
                KEventInfo dict_KEventInfo;
                KGenEventInfo dict_KGenEventInfo;
                KBasicJet dict_KBasicJet;
                KBasicJets dict_KBasicJets;
                KBasicTau dict_KBasicTau;
                KBasicTaus dict_KBasicTaus;
                KBeamSpot dict_KBeamSpot;
                KBeamSpots dict_KBeamSpots;
                KCaloJet dict_KCaloJet;
                KCaloJets dict_KCaloJets;
                KElectron dict_KElectron;
                KElectrons dict_KElectrons;
                KLV dict_KLV;
                KLVs dict_KLVs;
                KGenJet dict_KGenJet;
                KGenJets dict_KGenJets;
                KGenParticle dict_KGenParticle;
                KGenParticles dict_KGenParticles;
                KGenPhoton dict_KGenPhoton;
                KGenPhotons dict_KGenPhotons;
                KGenTau dict_KGenTau;
                KGenTaus dict_KGenTaus;
                KHit dict_KHit;
                KHits dict_KHits;
                KJet dict_KJet;
                KJets dict_KJets;
                KL1Muon dict_KL1Muon;
                KL1Muons dict_KL1Muons;
                KLepton dict_KLepton;
                KLeptons dict_KLeptons;
                KLeptonPair dict_KLeptonPair;
                KLeptonPairs dict_KLeptonPairs;
                KMET dict_KMET;
                KMETs dict_KMETs;
                KMuon dict_KMuon;
                KMuons dict_KMuons;
                KMuonTriggerCandidate dict_KMuonTriggerCandidate;
                KMuonTriggerCandidates dict_KMuonTriggerCandidates;
                KParticle dict_KParticle;
                KParticles dict_KParticles;
                KPFCandidate dict_KPFCandidate;
                KPFCandidates dict_KPFCandidates;
                KPhoton dict_KPhoton;
                KPhotons dict_KPhotons;
                KPileupDensity dict_KPileupDensity;
                KTau dict_KTau;
                KTaus dict_KTaus;
                KExtendedTau dict_KExtendedTau;
                KExtendedTaus dict_KExtendedTaus;
                KTaupairVerticesMap dict_KTaupairVerticesMap;
                KTaupairVerticesMaps dict_KTaupairVerticesMaps;
                KTrack dict_KTrack;
                KTracks dict_KTracks;
                KTriggerObjects dict_KTriggerObjects;
                KVertex dict_KVertex;
                KVertices dict_KVertices;
                KFilterSummary dict_KFilterSummary;
                KHCALNoiseSummary dict_KHCALNoiseSummary;
                KTrackSummary dict_KTrackSummary;
                KVertexSummary dict_KVertexSummary;
                KLumiInfo dict_KLumiInfo;
                KGenLumiInfo dict_KGenLumiInfo;
                KDataLumiInfo dict_KDataLumiInfo;
                KElectronMetadata dict_KElectronMetadata;
                KFilterMetadata dict_KFilterMetadata;
                KMuonMetadata dict_KMuonMetadata;
                KGenEventInfoMetadata dict_KGenEventInfoMetadata;
                KJetMetadata dict_KJetMetadata;
                KTauMetadata dict_KTauMetadata;
                KTriggerObjectMetadata dict_KTriggerObjectMetadata;
                KProvenance dict_KProvenance;
        };
}

#endif

#undef  _BACKWARD_BACKWARD_WARNING_H

input_line_221:65:10: fatal error: 'Kappa/DataFormats/interface/Hash.h' file not found
#include "Kappa/DataFormats/interface/Hash.h"
         ^
Error in TInterpreter::AutoParse: Error parsing payload code for class KTau with content:

#ifndef G__VECTOR_HAS_CLASS_ITERATOR
  #define G__VECTOR_HAS_CLASS_ITERATOR 1
#endif
#ifndef CMS_DICT_IMPL
  #define CMS_DICT_IMPL 1
#endif
#ifndef _REENTRANT
  #define _REENTRANT 1
#endif
#ifndef GNUSOURCE
  #define GNUSOURCE 1
#endif
#ifndef __STRICT_ANSI__
  #define **STRICT_ANSI** 1
#endif
#ifndef GNU_GCC
  #define GNU_GCC 1
#endif
#ifndef _GNU_SOURCE
  #define _GNU_SOURCE 1
#endif
#ifndef CMSSW_GIT_HASH
  #define CMSSW_GIT_HASH "CMSSW_7_4_7"
#endif
#ifndef PROJECT_NAME
  #define PROJECT_NAME "CMSSW"
#endif
#ifndef PROJECT_VERSION
  #define PROJECT_VERSION "CMSSW_7_4_7"
#endif
#ifndef BOOST_SPIRIT_THREADSAFE
  #define BOOST_SPIRIT_THREADSAFE 1
#endif
#ifndef PHOENIX_THREADSAFE
  #define PHOENIX_THREADSAFE 1
#endif
#ifndef CMSSW_REFLEX_DICT
  #define CMSSW_REFLEX_DICT 1
#endif

#define _BACKWARD_BACKWARD_WARNING_H
/\* This file is autogenerated by classes.UP. Do not edit!

```
Copyright (c) 2010 - 2015, All Rights Reserved
The authors are listed in the AUTHORS file.

ROOT dictionary that Kappa uses for reading the Kappa skim

    http://root.cern.ch/drupal/content/selecting-dictionary-entries-linkdefh
    http://root.cern.ch/drupal/content/interacting-shared-libraries-rootcint
```

*/

#ifndef KDATAFORMAT_CLASSES_H
#define KDATAFORMAT_CLASSES_H

#define G__DICTIONARY

#include "Kappa/DataFormats/interface/Hash.h"
#include "Kappa/DataFormats/interface/KBasic.h"
#include "Kappa/DataFormats/interface/KElectron.h"
#include "Kappa/DataFormats/interface/KInfo.h"
#include "Kappa/DataFormats/interface/KJetMET.h"
#include "Kappa/DataFormats/interface/KLepton.h"
#include "Kappa/DataFormats/interface/KMuon.h"
#include "Kappa/DataFormats/interface/KParticle.h"
#include "Kappa/DataFormats/interface/KPhoton.h"
#include "Kappa/DataFormats/interface/KRoot.h"
#include "Kappa/DataFormats/interface/KTau.h"
#include "Kappa/DataFormats/interface/KTrack.h"

namespace
{
        struct dictionary
        {
                RMFLV dict_RMFLV;
                RMFLV_Store dict_RMFLV_Store;
                RMDLV dict_RMDLV;
                RMDLV_Store dict_RMDLV_Store;
                KEventInfo dict_KEventInfo;
                KGenEventInfo dict_KGenEventInfo;
                KBasicJet dict_KBasicJet;
                KBasicJets dict_KBasicJets;
                KBasicTau dict_KBasicTau;
                KBasicTaus dict_KBasicTaus;
                KBeamSpot dict_KBeamSpot;
                KBeamSpots dict_KBeamSpots;
                KCaloJet dict_KCaloJet;
                KCaloJets dict_KCaloJets;
                KElectron dict_KElectron;
                KElectrons dict_KElectrons;
                KLV dict_KLV;
                KLVs dict_KLVs;
                KGenJet dict_KGenJet;
                KGenJets dict_KGenJets;
                KGenParticle dict_KGenParticle;
                KGenParticles dict_KGenParticles;
                KGenPhoton dict_KGenPhoton;
                KGenPhotons dict_KGenPhotons;
                KGenTau dict_KGenTau;
                KGenTaus dict_KGenTaus;
                KHit dict_KHit;
                KHits dict_KHits;
                KJet dict_KJet;
                KJets dict_KJets;
                KL1Muon dict_KL1Muon;
                KL1Muons dict_KL1Muons;
                KLepton dict_KLepton;
                KLeptons dict_KLeptons;
                KLeptonPair dict_KLeptonPair;
                KLeptonPairs dict_KLeptonPairs;
                KMET dict_KMET;
                KMETs dict_KMETs;
                KMuon dict_KMuon;
                KMuons dict_KMuons;
                KMuonTriggerCandidate dict_KMuonTriggerCandidate;
                KMuonTriggerCandidates dict_KMuonTriggerCandidates;
                KParticle dict_KParticle;
                KParticles dict_KParticles;
                KPFCandidate dict_KPFCandidate;
                KPFCandidates dict_KPFCandidates;
                KPhoton dict_KPhoton;
                KPhotons dict_KPhotons;
                KPileupDensity dict_KPileupDensity;
                KTau dict_KTau;
                KTaus dict_KTaus;
                KExtendedTau dict_KExtendedTau;
                KExtendedTaus dict_KExtendedTaus;
                KTaupairVerticesMap dict_KTaupairVerticesMap;
                KTaupairVerticesMaps dict_KTaupairVerticesMaps;
                KTrack dict_KTrack;
                KTracks dict_KTracks;
                KTriggerObjects dict_KTriggerObjects;
                KVertex dict_KVertex;
                KVertices dict_KVertices;
                KFilterSummary dict_KFilterSummary;
                KHCALNoiseSummary dict_KHCALNoiseSummary;
                KTrackSummary dict_KTrackSummary;
                KVertexSummary dict_KVertexSummary;
                KLumiInfo dict_KLumiInfo;
                KGenLumiInfo dict_KGenLumiInfo;
                KDataLumiInfo dict_KDataLumiInfo;
                KElectronMetadata dict_KElectronMetadata;
                KFilterMetadata dict_KFilterMetadata;
                KMuonMetadata dict_KMuonMetadata;
                KGenEventInfoMetadata dict_KGenEventInfoMetadata;
                KJetMetadata dict_KJetMetadata;
                KTauMetadata dict_KTauMetadata;
                KTriggerObjectMetadata dict_KTriggerObjectMetadata;
                KProvenance dict_KProvenance;
        };
}

#endif

#undef  _BACKWARD_BACKWARD_WARNING_H

input_line_222:65:10: fatal error: 'Kappa/DataFormats/interface/Hash.h' file not found
#include "Kappa/DataFormats/interface/Hash.h"
         ^
Error in TInterpreter::AutoParse: Error parsing payload code for class KGenTau with content:

#ifndef G__VECTOR_HAS_CLASS_ITERATOR
  #define G__VECTOR_HAS_CLASS_ITERATOR 1
#endif
#ifndef CMS_DICT_IMPL
  #define CMS_DICT_IMPL 1
#endif
#ifndef _REENTRANT
  #define _REENTRANT 1
#endif
#ifndef GNUSOURCE
  #define GNUSOURCE 1
#endif
#ifndef __STRICT_ANSI__
  #define **STRICT_ANSI** 1
#endif
#ifndef GNU_GCC
  #define GNU_GCC 1
#endif
#ifndef _GNU_SOURCE
  #define _GNU_SOURCE 1
#endif
#ifndef CMSSW_GIT_HASH
  #define CMSSW_GIT_HASH "CMSSW_7_4_7"
#endif
#ifndef PROJECT_NAME
  #define PROJECT_NAME "CMSSW"
#endif
#ifndef PROJECT_VERSION
  #define PROJECT_VERSION "CMSSW_7_4_7"
#endif
#ifndef BOOST_SPIRIT_THREADSAFE
  #define BOOST_SPIRIT_THREADSAFE 1
#endif
#ifndef PHOENIX_THREADSAFE
  #define PHOENIX_THREADSAFE 1
#endif
#ifndef CMSSW_REFLEX_DICT
  #define CMSSW_REFLEX_DICT 1
#endif

#define _BACKWARD_BACKWARD_WARNING_H
/\* This file is autogenerated by classes.UP. Do not edit!

```
Copyright (c) 2010 - 2015, All Rights Reserved
The authors are listed in the AUTHORS file.

ROOT dictionary that Kappa uses for reading the Kappa skim

    http://root.cern.ch/drupal/content/selecting-dictionary-entries-linkdefh
    http://root.cern.ch/drupal/content/interacting-shared-libraries-rootcint
```

*/

#ifndef KDATAFORMAT_CLASSES_H
#define KDATAFORMAT_CLASSES_H

#define G__DICTIONARY

#include "Kappa/DataFormats/interface/Hash.h"
#include "Kappa/DataFormats/interface/KBasic.h"
#include "Kappa/DataFormats/interface/KElectron.h"
#include "Kappa/DataFormats/interface/KInfo.h"
#include "Kappa/DataFormats/interface/KJetMET.h"
#include "Kappa/DataFormats/interface/KLepton.h"
#include "Kappa/DataFormats/interface/KMuon.h"
#include "Kappa/DataFormats/interface/KParticle.h"
#include "Kappa/DataFormats/interface/KPhoton.h"
#include "Kappa/DataFormats/interface/KRoot.h"
#include "Kappa/DataFormats/interface/KTau.h"
#include "Kappa/DataFormats/interface/KTrack.h"

namespace
{
        struct dictionary
        {
                RMFLV dict_RMFLV;
                RMFLV_Store dict_RMFLV_Store;
                RMDLV dict_RMDLV;
                RMDLV_Store dict_RMDLV_Store;
                KEventInfo dict_KEventInfo;
                KGenEventInfo dict_KGenEventInfo;
                KBasicJet dict_KBasicJet;
                KBasicJets dict_KBasicJets;
                KBasicTau dict_KBasicTau;
                KBasicTaus dict_KBasicTaus;
                KBeamSpot dict_KBeamSpot;
                KBeamSpots dict_KBeamSpots;
                KCaloJet dict_KCaloJet;
                KCaloJets dict_KCaloJets;
                KElectron dict_KElectron;
                KElectrons dict_KElectrons;
                KLV dict_KLV;
                KLVs dict_KLVs;
                KGenJet dict_KGenJet;
                KGenJets dict_KGenJets;
                KGenParticle dict_KGenParticle;
                KGenParticles dict_KGenParticles;
                KGenPhoton dict_KGenPhoton;
                KGenPhotons dict_KGenPhotons;
                KGenTau dict_KGenTau;
                KGenTaus dict_KGenTaus;
                KHit dict_KHit;
                KHits dict_KHits;
                KJet dict_KJet;
                KJets dict_KJets;
                KL1Muon dict_KL1Muon;
                KL1Muons dict_KL1Muons;
                KLepton dict_KLepton;
                KLeptons dict_KLeptons;
                KLeptonPair dict_KLeptonPair;
                KLeptonPairs dict_KLeptonPairs;
                KMET dict_KMET;
                KMETs dict_KMETs;
                KMuon dict_KMuon;
                KMuons dict_KMuons;
                KMuonTriggerCandidate dict_KMuonTriggerCandidate;
                KMuonTriggerCandidates dict_KMuonTriggerCandidates;
                KParticle dict_KParticle;
                KParticles dict_KParticles;
                KPFCandidate dict_KPFCandidate;
                KPFCandidates dict_KPFCandidates;
                KPhoton dict_KPhoton;
                KPhotons dict_KPhotons;
                KPileupDensity dict_KPileupDensity;
                KTau dict_KTau;
                KTaus dict_KTaus;
                KExtendedTau dict_KExtendedTau;
                KExtendedTaus dict_KExtendedTaus;
                KTaupairVerticesMap dict_KTaupairVerticesMap;
                KTaupairVerticesMaps dict_KTaupairVerticesMaps;
                KTrack dict_KTrack;
                KTracks dict_KTracks;
                KTriggerObjects dict_KTriggerObjects;
                KVertex dict_KVertex;
                KVertices dict_KVertices;
                KFilterSummary dict_KFilterSummary;
                KHCALNoiseSummary dict_KHCALNoiseSummary;
                KTrackSummary dict_KTrackSummary;
                KVertexSummary dict_KVertexSummary;
                KLumiInfo dict_KLumiInfo;
                KGenLumiInfo dict_KGenLumiInfo;
                KDataLumiInfo dict_KDataLumiInfo;
                KElectronMetadata dict_KElectronMetadata;
                KFilterMetadata dict_KFilterMetadata;
                KMuonMetadata dict_KMuonMetadata;
                KGenEventInfoMetadata dict_KGenEventInfoMetadata;
                KJetMetadata dict_KJetMetadata;
                KTauMetadata dict_KTauMetadata;
                KTriggerObjectMetadata dict_KTriggerObjectMetadata;
                KProvenance dict_KProvenance;
        };
}

#endif

#undef  _BACKWARD_BACKWARD_WARNING_H

input_line_223:65:10: fatal error: 'Kappa/DataFormats/interface/Hash.h' file not found
#include "Kappa/DataFormats/interface/Hash.h"
         ^
Error in TInterpreter::AutoParse: Error parsing payload code for class KGenJet with content:

#ifndef G__VECTOR_HAS_CLASS_ITERATOR
  #define G__VECTOR_HAS_CLASS_ITERATOR 1
#endif
#ifndef CMS_DICT_IMPL
  #define CMS_DICT_IMPL 1
#endif
#ifndef _REENTRANT
  #define _REENTRANT 1
#endif
#ifndef GNUSOURCE
  #define GNUSOURCE 1
#endif
#ifndef __STRICT_ANSI__
  #define **STRICT_ANSI** 1
#endif
#ifndef GNU_GCC
  #define GNU_GCC 1
#endif
#ifndef _GNU_SOURCE
  #define _GNU_SOURCE 1
#endif
#ifndef CMSSW_GIT_HASH
  #define CMSSW_GIT_HASH "CMSSW_7_4_7"
#endif
#ifndef PROJECT_NAME
  #define PROJECT_NAME "CMSSW"
#endif
#ifndef PROJECT_VERSION
  #define PROJECT_VERSION "CMSSW_7_4_7"
#endif
#ifndef BOOST_SPIRIT_THREADSAFE
  #define BOOST_SPIRIT_THREADSAFE 1
#endif
#ifndef PHOENIX_THREADSAFE
  #define PHOENIX_THREADSAFE 1
#endif
#ifndef CMSSW_REFLEX_DICT
  #define CMSSW_REFLEX_DICT 1
#endif

#define _BACKWARD_BACKWARD_WARNING_H
/\* This file is autogenerated by classes.UP. Do not edit!

```
Copyright (c) 2010 - 2015, All Rights Reserved
The authors are listed in the AUTHORS file.

ROOT dictionary that Kappa uses for reading the Kappa skim

    http://root.cern.ch/drupal/content/selecting-dictionary-entries-linkdefh
    http://root.cern.ch/drupal/content/interacting-shared-libraries-rootcint
```

*/

#ifndef KDATAFORMAT_CLASSES_H
#define KDATAFORMAT_CLASSES_H

#define G__DICTIONARY

#include "Kappa/DataFormats/interface/Hash.h"
#include "Kappa/DataFormats/interface/KBasic.h"
#include "Kappa/DataFormats/interface/KElectron.h"
#include "Kappa/DataFormats/interface/KInfo.h"
#include "Kappa/DataFormats/interface/KJetMET.h"
#include "Kappa/DataFormats/interface/KLepton.h"
#include "Kappa/DataFormats/interface/KMuon.h"
#include "Kappa/DataFormats/interface/KParticle.h"
#include "Kappa/DataFormats/interface/KPhoton.h"
#include "Kappa/DataFormats/interface/KRoot.h"
#include "Kappa/DataFormats/interface/KTau.h"
#include "Kappa/DataFormats/interface/KTrack.h"

namespace
{
        struct dictionary
        {
                RMFLV dict_RMFLV;
                RMFLV_Store dict_RMFLV_Store;
                RMDLV dict_RMDLV;
                RMDLV_Store dict_RMDLV_Store;
                KEventInfo dict_KEventInfo;
                KGenEventInfo dict_KGenEventInfo;
                KBasicJet dict_KBasicJet;
                KBasicJets dict_KBasicJets;
                KBasicTau dict_KBasicTau;
                KBasicTaus dict_KBasicTaus;
                KBeamSpot dict_KBeamSpot;
                KBeamSpots dict_KBeamSpots;
                KCaloJet dict_KCaloJet;
                KCaloJets dict_KCaloJets;
                KElectron dict_KElectron;
                KElectrons dict_KElectrons;
                KLV dict_KLV;
                KLVs dict_KLVs;
                KGenJet dict_KGenJet;
                KGenJets dict_KGenJets;
                KGenParticle dict_KGenParticle;
                KGenParticles dict_KGenParticles;
                KGenPhoton dict_KGenPhoton;
                KGenPhotons dict_KGenPhotons;
                KGenTau dict_KGenTau;
                KGenTaus dict_KGenTaus;
                KHit dict_KHit;
                KHits dict_KHits;
                KJet dict_KJet;
                KJets dict_KJets;
                KL1Muon dict_KL1Muon;
                KL1Muons dict_KL1Muons;
                KLepton dict_KLepton;
                KLeptons dict_KLeptons;
                KLeptonPair dict_KLeptonPair;
                KLeptonPairs dict_KLeptonPairs;
                KMET dict_KMET;
                KMETs dict_KMETs;
                KMuon dict_KMuon;
                KMuons dict_KMuons;
                KMuonTriggerCandidate dict_KMuonTriggerCandidate;
                KMuonTriggerCandidates dict_KMuonTriggerCandidates;
                KParticle dict_KParticle;
                KParticles dict_KParticles;
                KPFCandidate dict_KPFCandidate;
                KPFCandidates dict_KPFCandidates;
                KPhoton dict_KPhoton;
                KPhotons dict_KPhotons;
                KPileupDensity dict_KPileupDensity;
                KTau dict_KTau;
                KTaus dict_KTaus;
                KExtendedTau dict_KExtendedTau;
                KExtendedTaus dict_KExtendedTaus;
                KTaupairVerticesMap dict_KTaupairVerticesMap;
                KTaupairVerticesMaps dict_KTaupairVerticesMaps;
                KTrack dict_KTrack;
                KTracks dict_KTracks;
                KTriggerObjects dict_KTriggerObjects;
                KVertex dict_KVertex;
                KVertices dict_KVertices;
                KFilterSummary dict_KFilterSummary;
                KHCALNoiseSummary dict_KHCALNoiseSummary;
                KTrackSummary dict_KTrackSummary;
                KVertexSummary dict_KVertexSummary;
                KLumiInfo dict_KLumiInfo;
                KGenLumiInfo dict_KGenLumiInfo;
                KDataLumiInfo dict_KDataLumiInfo;
                KElectronMetadata dict_KElectronMetadata;
                KFilterMetadata dict_KFilterMetadata;
                KMuonMetadata dict_KMuonMetadata;
                KGenEventInfoMetadata dict_KGenEventInfoMetadata;
                KJetMetadata dict_KJetMetadata;
                KTauMetadata dict_KTauMetadata;
                KTriggerObjectMetadata dict_KTriggerObjectMetadata;
                KProvenance dict_KProvenance;
        };
}

#endif

#undef  _BACKWARD_BACKWARD_WARNING_H

input_line_224:65:10: fatal error: 'Kappa/DataFormats/interface/Hash.h' file not found
#include "Kappa/DataFormats/interface/Hash.h"
         ^
Error in TInterpreter::AutoParse: Error parsing payload code for class KJet with content:

#ifndef G__VECTOR_HAS_CLASS_ITERATOR
  #define G__VECTOR_HAS_CLASS_ITERATOR 1
#endif
#ifndef CMS_DICT_IMPL
  #define CMS_DICT_IMPL 1
#endif
#ifndef _REENTRANT
  #define _REENTRANT 1
#endif
#ifndef GNUSOURCE
  #define GNUSOURCE 1
#endif
#ifndef __STRICT_ANSI__
  #define **STRICT_ANSI** 1
#endif
#ifndef GNU_GCC
  #define GNU_GCC 1
#endif
#ifndef _GNU_SOURCE
  #define _GNU_SOURCE 1
#endif
#ifndef CMSSW_GIT_HASH
  #define CMSSW_GIT_HASH "CMSSW_7_4_7"
#endif
#ifndef PROJECT_NAME
  #define PROJECT_NAME "CMSSW"
#endif
#ifndef PROJECT_VERSION
  #define PROJECT_VERSION "CMSSW_7_4_7"
#endif
#ifndef BOOST_SPIRIT_THREADSAFE
  #define BOOST_SPIRIT_THREADSAFE 1
#endif
#ifndef PHOENIX_THREADSAFE
  #define PHOENIX_THREADSAFE 1
#endif
#ifndef CMSSW_REFLEX_DICT
  #define CMSSW_REFLEX_DICT 1
#endif

#define _BACKWARD_BACKWARD_WARNING_H
/\* This file is autogenerated by classes.UP. Do not edit!

```
Copyright (c) 2010 - 2015, All Rights Reserved
The authors are listed in the AUTHORS file.

ROOT dictionary that Kappa uses for reading the Kappa skim

    http://root.cern.ch/drupal/content/selecting-dictionary-entries-linkdefh
    http://root.cern.ch/drupal/content/interacting-shared-libraries-rootcint
```

*/

#ifndef KDATAFORMAT_CLASSES_H
#define KDATAFORMAT_CLASSES_H

#define G__DICTIONARY

#include "Kappa/DataFormats/interface/Hash.h"
#include "Kappa/DataFormats/interface/KBasic.h"
#include "Kappa/DataFormats/interface/KElectron.h"
#include "Kappa/DataFormats/interface/KInfo.h"
#include "Kappa/DataFormats/interface/KJetMET.h"
#include "Kappa/DataFormats/interface/KLepton.h"
#include "Kappa/DataFormats/interface/KMuon.h"
#include "Kappa/DataFormats/interface/KParticle.h"
#include "Kappa/DataFormats/interface/KPhoton.h"
#include "Kappa/DataFormats/interface/KRoot.h"
#include "Kappa/DataFormats/interface/KTau.h"
#include "Kappa/DataFormats/interface/KTrack.h"

namespace
{
        struct dictionary
        {
                RMFLV dict_RMFLV;
                RMFLV_Store dict_RMFLV_Store;
                RMDLV dict_RMDLV;
                RMDLV_Store dict_RMDLV_Store;
                KEventInfo dict_KEventInfo;
                KGenEventInfo dict_KGenEventInfo;
                KBasicJet dict_KBasicJet;
                KBasicJets dict_KBasicJets;
                KBasicTau dict_KBasicTau;
                KBasicTaus dict_KBasicTaus;
                KBeamSpot dict_KBeamSpot;
                KBeamSpots dict_KBeamSpots;
                KCaloJet dict_KCaloJet;
                KCaloJets dict_KCaloJets;
                KElectron dict_KElectron;
                KElectrons dict_KElectrons;
                KLV dict_KLV;
                KLVs dict_KLVs;
                KGenJet dict_KGenJet;
                KGenJets dict_KGenJets;
                KGenParticle dict_KGenParticle;
                KGenParticles dict_KGenParticles;
                KGenPhoton dict_KGenPhoton;
                KGenPhotons dict_KGenPhotons;
                KGenTau dict_KGenTau;
                KGenTaus dict_KGenTaus;
                KHit dict_KHit;
                KHits dict_KHits;
                KJet dict_KJet;
                KJets dict_KJets;
                KL1Muon dict_KL1Muon;
                KL1Muons dict_KL1Muons;
                KLepton dict_KLepton;
                KLeptons dict_KLeptons;
                KLeptonPair dict_KLeptonPair;
                KLeptonPairs dict_KLeptonPairs;
                KMET dict_KMET;
                KMETs dict_KMETs;
                KMuon dict_KMuon;
                KMuons dict_KMuons;
                KMuonTriggerCandidate dict_KMuonTriggerCandidate;
                KMuonTriggerCandidates dict_KMuonTriggerCandidates;
                KParticle dict_KParticle;
                KParticles dict_KParticles;
                KPFCandidate dict_KPFCandidate;
                KPFCandidates dict_KPFCandidates;
                KPhoton dict_KPhoton;
                KPhotons dict_KPhotons;
                KPileupDensity dict_KPileupDensity;
                KTau dict_KTau;
                KTaus dict_KTaus;
                KExtendedTau dict_KExtendedTau;
                KExtendedTaus dict_KExtendedTaus;
                KTaupairVerticesMap dict_KTaupairVerticesMap;
                KTaupairVerticesMaps dict_KTaupairVerticesMaps;
                KTrack dict_KTrack;
                KTracks dict_KTracks;
                KTriggerObjects dict_KTriggerObjects;
                KVertex dict_KVertex;
                KVertices dict_KVertices;
                KFilterSummary dict_KFilterSummary;
                KHCALNoiseSummary dict_KHCALNoiseSummary;
                KTrackSummary dict_KTrackSummary;
                KVertexSummary dict_KVertexSummary;
                KLumiInfo dict_KLumiInfo;
                KGenLumiInfo dict_KGenLumiInfo;
                KDataLumiInfo dict_KDataLumiInfo;
                KElectronMetadata dict_KElectronMetadata;
                KFilterMetadata dict_KFilterMetadata;
                KMuonMetadata dict_KMuonMetadata;
                KGenEventInfoMetadata dict_KGenEventInfoMetadata;
                KJetMetadata dict_KJetMetadata;
                KTauMetadata dict_KTauMetadata;
                KTriggerObjectMetadata dict_KTriggerObjectMetadata;
                KProvenance dict_KProvenance;
        };
}

#endif

#undef  _BACKWARD_BACKWARD_WARNING_H

Dictionary for class KMET is not consistent with file content!
        Branch content: {chargedHadronFraction, electronFraction, hfEMFraction, hfHadronFraction, leptonSelectionHash, muonFraction, neutralHadronFraction, p4.fCoordinates.fEta, p4.fCoordinates.fM, p4.fCoordinates.fPhi, p4.fCoordinates.fPt, photonFraction, significance.fRep.fArray[3], sumEt}

```
      Dict content: {chargedHadronFraction, electronFraction, hfEMFraction, hfHadronFraction, leptonSelectionHash, muonFraction, neutralHadronFraction, photonFraction, significance.fRep.fArray[3], sumEt}
```

FATAL: Requested branch (genmetTrue) not found!
WARN : Aborting application. Reason: Fatal log at [/.automount/home/home__home2/institut_3b/tmuller/home/cms/htt/analysis/CMSSW_7_4_7/src/Artus/KappaAnalysis/interface/KappaEventProviderBase.h:120]
ERROR: Exit with code -6.

thomas-mueller commented:
This is the error I get in case of batch jobs. Locally it works.

swayand commented:
Hi Thomas,

Localy it works but on the batch system not. My first guess would be that the gc configs are wrong configurations. Therefor just have a look at the gc config in the work directory.
And especially look at manipulations of Ld_path or so.

I will have a look at this problem by tomorrow.

Cheers Stefan

---- Thomas Müller schrieb ----

This is the error I get in case of batch jobs. Locally it works.

—
You are receiving this because you modified the open/close state.
Reply to this email directly, view it on GitHubhttps://github.com/KappaAnalysis/Kappa/pull/31#issuecomment-247376326, or mute the threadhttps://github.com/notifications/unsubscribe-auth/AGP8_j4CSl-8O8Fs5glFyXO9TM0bdtOxks5qqW_DgaJpZM4J6p6t.

thomas-mueller commented:
It seems, that the compiled binary still depends on the header files, which is not too clear for me. At least [this commit](https://github.com/artus-analysis/Artus/commit/a1279f17c89d975b93c54f53d65d5d2aaa6fe783) seems to fix my problem and I can submit the jobs successfully. I did not write earlier, because the control plots look very different to my last ones and I am still trying to find out the reason.
